src/mono/mono/metadata/metadata-internals.h

```diff
@@ -431,6 +431,7 @@ struct _MonoImage {
 	 */
 	MonoAssembly **references;
 	int nreferences;
+	mono_mutex_t references_lock;
```

Comment:
I wonder what it would take to split up `MonoImage` into a base class (say `MonoEcmaImage`) that only knows about tables and such and make the current `MonoImage` a subclass for the main image of an assembly.  And we could have a separate `MonoPdbImage` or `MonoDeltaImage` that only has a subset of the concrete fields that are relevant for PPDBs or E&C images.